app/controllers/nonsipp/loansmadeoroutstanding/LoansCYAController.scala

```diff
@@ -0,0 +1,434 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.nonsipp.loansmadeoroutstanding
+
+import cats.implicits.toShow
+import config.Refined.Max9999999
+import controllers.actions._
+import controllers.nonsipp.loansmadeoroutstanding.LoansCYAController._
+import controllers.PSRController
+import models.SchemeId.Srn
+import models._
+import navigation.Navigator
+import pages.nonsipp.loansmadeoroutstanding._
+import play.api.i18n._
+import play.api.mvc._
+import services.SchemeDateService
+import utils.DateTimeUtils.localDateShow
+import utils.ListUtils.ListOps
+import viewmodels.DisplayMessage._
+import viewmodels.implicits._
+import viewmodels.models._
+import views.html.CheckYourAnswersView
+import utils.Tuple2Utils._
+
+import java.time.LocalDate
+import javax.inject.{Inject, Named}
+
+class LoansCYAController @Inject()(
+  override val messagesApi: MessagesApi,
+  @Named("non-sipp") navigator: Navigator,
+  identifyAndRequireData: IdentifyAndRequireData,
+  val controllerComponents: MessagesControllerComponents,
+  schemeDateService: SchemeDateService,
+  view: CheckYourAnswersView
```

Comment:
Are we expecting the navigation to come back to the cya page after a change link was pressed and the individual page submitted?

Reply:
It's not mentioned in the ticket but it would require a lot of extra work in certain scenarios e.g. if you went to change the who received loan question to something else and it deletes all the user answers for the loans journey you can't jump back to the cya page